homeassistant/components/sensor/elkm1.py

```diff
@@ -0,0 +1,315 @@
+
+    def __init__(self, element, elk, elk_data):
+        """Initialize the base of all Elk sensors."""
+        super().__init__(element, elk, elk_data)
+        self._state = None
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+
+class ElkCounter(ElkSensor):
+    """Representation of an Elk-M1 Counter."""
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return 'mdi:numeric'
+
+    def _element_changed(self, element, changeset):
+        self._state = self._element.value
+
+
+class ElkKeypad(ElkSensor):
+    """Representation of an Elk-M1 Keypad."""
+
+    @property
+    def temperature_unit(self):
+        """Return the temperature unit."""
+        return self._temperature_unit
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._temperature_unit
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return 'mdi:thermometer-lines'
+
+    @property
+    def device_state_attributes(self):
+        """Attributes of the sensor."""
+        from elkm1_lib.util import username
+
+        attrs = self.initial_attrs()
+        attrs['area'] = self._element.area + 1
+        attrs['temperature'] = self._element.temperature
+        attrs['last_user_time'] = self._element.last_user_time.isoformat()
+        attrs['last_user'] = self._element.last_user + 1
+        attrs['code'] = self._element.code
+        attrs['last_user_name'] = username(self._elk, self._element.last_user)
+        return attrs
+
+    def _element_changed(self, element, changeset):
+        self._state = temperature_to_state(self._element.temperature, -40)
+
+    async def async_added_to_hass(self):
+        """Register callback for ElkM1 changes and update entity state."""
+        await super().async_added_to_hass()
+        self.hass.data[ELK_DOMAIN]['keypads'][
+            self._element.index] = self.entity_id
+
+
+class ElkPanel(ElkSensor):
+    """Representation of an Elk-M1 Panel."""
+
+    async def async_added_to_hass(self):
+        """Register callback for ElkM1 changes."""
+        await super().async_added_to_hass()
+        async_dispatcher_connect(
+            self.hass, '{}_{}'.format(SIGNAL_SPEAK_WORD, self.entity_id),
+            self._speak_word)
+        async_dispatcher_connect(
+            self.hass, '{}_{}'.format(SIGNAL_SPEAK_PHRASE, self.entity_id),
+            self._speak_phrase)
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return "mdi:home"
+
+    @property
+    def device_state_attributes(self):
+        """Attributes of the sensor."""
+        attrs = self.initial_attrs()
+        attrs['elkm1_version'] = self._element.elkm1_version
+        attrs['system_trouble_status'] = self._element.system_trouble_status
+        attrs['xep_version'] = self._element.xep_version
```

Comment:
What is xep_version?

Reply:
It is the version of the Ethernet controller.

Reply:
Ok, then it has to be removed too.